shared/src/metabase/shared/util/time.cljc

```diff
@@ -0,0 +1,255 @@
+(ns metabase.shared.util.time
+  "Time parsing helper functions.
+  In Java these return [[OffsetDateTime]], in JavaScript they return Moments."
+  (:require
+   [clojure.string :as str]
+   [metabase.shared.util :as shared.u]
+   #?@(:clj  [[java-time :as t]
+              [metabase.public-settings :as public-settings]]
+       :cljs [["moment" :as moment]]))
+  #?(:clj (:import
+           java.util.Locale)))
```

Comment:
It seems like almost everything in this namespace has a different Clj vs. Cljs implementation. Would it make sense to just do separate Clj/Cljs versions of the namespace? So you don't need to use so much conditional logic.